Cabal/src/Distribution/Simple/Program/GHC.hs

```diff
@@ -329,6 +329,7 @@ normaliseGhcArgs (Just ghcVersion) PackageDescription{..} ghcArgs
           ]
         , from [8, 2] ["-fmax-uncovered-patterns", "-fmax-errors"]
         , from [8, 4] $ to [8, 6] ["-fmax-valid-substitutions"]
+        , from [8, 12] ["-fmax-forced-spec-args", "-fwrite-if-compression"]
```

Comment:
It's probably a dumb question but: why is it `from [8, 12]` while the changelog talks about 9.12?

Reply:
I incorrecvctly was looking at the previous line when writing that. In any case, this is still draft and I should label it so: I need to check the oother options, and will probably have to spelunk the hash generation code to figure out how to deal with @gulinserge's new options (/cc: @mpilgrem, stack probably needs to deal with them as well).

Reply:
On second thought, see comment below re the new options.